pkg/controller/common/certificates/ca.go

```diff
@@ -124,3 +130,18 @@ func (c *CA) CreateCertificate(
 
 	return certData, err
 }
+
+// HasPublicCA returns true if an Elastic resource has a CA (ca.crt) in its public HTTP certs secret given its namer,
+// namespace and name.
+func HasPublicCA(client k8s.Client, namer name.Namer, namespace string, name string) (bool, error) {
```

Comment:
nit, naming: `ca.crt` can be managed in the context of a private PKI, without being a public CA

```suggestion
// PublicCertsHasCACert returns true if an Elastic resource has a Certificate Authority certificate (ca.crt) in its public HTTP certs secret given its namer,
// namespace and name.
func PublicCertsHasCACert(client k8s.Client, namer name.Namer, namespace string, name string) (bool, error) {
```